crates/ethereum/evm/src/execute.rs

```diff
@@ -426,7 +451,38 @@ where
                 Some(p) => p,
             };
 
-            let bitcoin_checkpoint = consensus_pkg.bitcoin_checkpoint;
+            // Get the reference block hash from the pegin metadata. 
+            // This is used to avoid the growing list of headers in the pegin metadata
+            // by using a bitcoin checkpoint that is close to the pegin block height. 
+            // The reference block hash is only provided for version v1.
+            if pegin_data.meta[0].version == 1 && pegin_data.meta[0].ref_block_hash.is_none() {
+                return Err(MintContractError::InvalidPeginData {
+                    error: format!("Reference block hash not found"),
+                    revert_address: pegin_data.account,
+                    revert_amount: pegin_data.amount,
+                });
+            }
+
+            let bitcoin_checkpoint = if let Some(ref_block_hash) = pegin_data.meta[0].ref_block_hash {
+                if let Some(block) = provider.find_block_by_hash(ref_block_hash, reth_provider::BlockSource::Any).map_err(|_| MintContractError::InvalidPeginData {
+                    error: format!("Block not found"),
```

Comment:
Lets be more specific.
```suggestion
                    error: format!("refrence block hash not found"),
```

Reply:
Done!